artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRaUtils.java

```diff
@@ -226,19 +226,18 @@ public static Object lookup(final Context context, final String name, final Clas
 
    /**
     * Within AS7 the RA is loaded by JCA. properties can only be passed in String form. However if
-    * RA is configured using jgroups stack, we need to pass a Channel object. As is impossible with
-    * JCA, we use this method to allow a JChannel object to be located.
+    * RA is configured using jgroups stack, we need to pass a BroadcastEndpointFactory object. As is impossible with
+    * JCA, we use this method to allow a BroadcastEndpointFactory object to be located.
     */
-   public static JChannel locateJGroupsChannel(final String locatorClass, final String name) {
-      return AccessController.doPrivileged(new PrivilegedAction<JChannel>() {
+   public static BroadcastEndpointFactory locateBroadcastEndpointFactory(final String locatorClass, final String name) {
+      return AccessController.doPrivileged(new PrivilegedAction<BroadcastEndpointFactory>() {
          @Override
-         public JChannel run() {
+         public BroadcastEndpointFactory run() {
             try {
-               ClassLoader loader = Thread.currentThread().getContextClassLoader();
-               Class<?> aClass = loader.loadClass(locatorClass);
+               Class<?> aClass = getClass().getClassLoader().loadClass(locatorClass);
                Object o = aClass.newInstance();
-               Method m = aClass.getMethod("locateChannel", new Class[]{String.class});
-               return (JChannel) m.invoke(o, name);
+               Method m = aClass.getMethod("locateBroadcastEndpointFactory", new Class[]{String.class});
```

Comment:
I worry about how this change affects compatibility.  Existing usage of the jgroupsChannelLocatorClass property will break, as the value provided will not have this method.
This change isn't explicitly required to fix https://issues.jboss.org/browse/WFLY-9246, so long as the messaging-activemq subsystem is able to override the broadcast endpoint logic used by the resource adapter logic, which https://github.com/rh-messaging/jboss-activemq-artemis/commit/1692f7818d9d97a1a9704740fab53f4798120f35 already allows.